modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectValidationRuleLocalServiceTest.java

```diff
@@ -69,31 +85,119 @@ public void testAddObjectValidationRule() throws Exception {
+				_getObjectValidationRuleSetting(
+					ObjectValidationRuleSettingConstants.NAME_OBJECT_FIELD_ID,
+					RandomTestUtil.randomString())),
+			ObjectValidationRuleConstants.OUTPUT_TYPE_FULL_VALIDATION,
+			_VALID_DDM_SCRIPT);
+		_testAddObjectValidationRuleFailure(
+			ObjectValidationRuleConstants.ENGINE_TYPE_DDM,
+			ObjectValidationRuleSettingNameException.MissingRequiredName.class,
+			String.format(
+				"The object validation rule setting \"%s\" is required",
+				ObjectValidationRuleSettingConstants.NAME_OBJECT_FIELD_ID),
+			RandomTestUtil.randomString(), Collections.emptyList(),
+			ObjectValidationRuleConstants.OUTPUT_TYPE_PARTIAL_VALIDATION,
+			_VALID_DDM_SCRIPT);
+
+		String randomString = RandomTestUtil.randomString();
+
+		_testAddObjectValidationRuleFailure(
+			ObjectValidationRuleConstants.ENGINE_TYPE_DDM,
+			ObjectValidationRuleSettingValueException.InvalidValue.class,
+			String.format(
+				"The value \"%s\" of the object validation rule setting " +
+					"\"%s\" is invalid",
+				randomString,
+				ObjectValidationRuleSettingConstants.NAME_OBJECT_FIELD_ID),
+			RandomTestUtil.randomString(),
+			Collections.singletonList(
+				_getObjectValidationRuleSetting(
+					ObjectValidationRuleSettingConstants.NAME_OBJECT_FIELD_ID,
+					randomString)),
+			ObjectValidationRuleConstants.OUTPUT_TYPE_PARTIAL_VALIDATION,
+			_VALID_DDM_SCRIPT);
+
 		_testAddObjectValidationRuleFailure(
 			ObjectValidationRuleConstants.ENGINE_TYPE_GROOVY,
 			ObjectValidationRuleScriptException.class, "syntax-error",
-			RandomTestUtil.randomString(), "import;\ninvalidFields = false;");
+			RandomTestUtil.randomString(), Collections.emptyList(),
+			ObjectValidationRuleConstants.OUTPUT_TYPE_FULL_VALIDATION,
+			"import;\ninvalidFields = false;");
 		_testAddObjectValidationRuleFailure(
 			StringPool.BLANK,
 			ObjectValidationRuleEngineException.MustNotBeNull.class,
-			"Engine is null", RandomTestUtil.randomString(), _VALID_DDM_SCRIPT);
+			"Engine is null", RandomTestUtil.randomString(),
+			Collections.emptyList(),
+			ObjectValidationRuleConstants.OUTPUT_TYPE_FULL_VALIDATION,
+			_VALID_DDM_SCRIPT);
 		_testAddObjectValidationRuleFailure(
 			"abcdefghijklmnopqrstuvwxyz",
 			ObjectValidationRuleEngineException.NoSuchEngine.class,
 			"Engine \"abcdefghijklmnopqrstuvwxyz\" does not exist",
-			RandomTestUtil.randomString(), _VALID_DDM_SCRIPT);
+			RandomTestUtil.randomString(), Collections.emptyList(),
+			ObjectValidationRuleConstants.OUTPUT_TYPE_FULL_VALIDATION,
+			_VALID_DDM_SCRIPT);
 
 		_testAddObjectValidationRuleSuccess(
 			ObjectValidationRuleConstants.ENGINE_TYPE_DDM, _VALID_DDM_SCRIPT);
 		_testAddObjectValidationRuleSuccess(
 			ObjectValidationRuleConstants.ENGINE_TYPE_GROOVY,
 			"import com.liferay.commerce.service.CommerceOrderLocalService;\n" +
 				"invalidFields = false;");
+
+		ObjectField objectField = _objectFieldLocalService.getObjectFields(
+			_objectDefinition.getObjectDefinitionId(), false
+		).get(
+			0
+		);
```

Comment:
I believe that this code will not pass in the Brian's review. In this case is better make:
```java
List<ObjectField> objectFields =
			_objectFieldLocalService.getObjectFields(
				_objectDefinition.getObjectDefinitionId(), false);

ObjectField objectField = objectFields.get(0);
```